pydoctor/astutils.py

```diff
@@ -34,3 +35,26 @@ def bind_args(sig: Signature, call: ast.Call) -> BoundArguments:
         if kw.arg is not None
         }
     return sig.bind(*call.args, **kwargs)
+
+
+if sys.version_info[:2] >= (3, 8):
+    # Since Python 3.8 "foo" is parsed as ast.Constant.
+    def _is_str_constant(expr: ast.expr, s: str) -> bool:
+        return isinstance(expr, ast.Constant) \
+        and expr.value == '__main__'
+else:
+    # Before Python 3.8 "foo" was parsed as ast.Str.
+    def _is_str_constant(expr: ast.expr, s: str) -> bool:
+        return isinstance(expr, ast.Str) \
+        and expr.s == '__main__'
```

Comment:
Also it seem that the current implementation ignores the parameter `s`.

```suggestion
def _is_str_constant(expr: ast.expr, s: str) -> bool:
    try:
        value = ast.literal_eval(expr)
    except Exception:
        # literal_eval can raise a variety of exceptions...
        return False
    else:
        return value==s
```